packages/server-wallet/src/protocols/direct-funding.ts

```diff
@@ -1,6 +1,88 @@
-import {right} from 'fp-ts/lib/Either';
-import {none} from 'fp-ts/lib/Option';
+import {right, left} from 'fp-ts/lib/Either';
+import {none, some} from 'fp-ts/lib/Option';
+import {
+  SimpleAllocation,
+  BN,
+  Outcome,
+  Zero,
+  simpleEthAllocation,
+  isSimpleEthAllocation,
+  simpleTokenAllocation,
+} from '@statechannels/wallet-core';
 
-import {Protocol, ChannelState} from './state';
+import {match} from '../match';
 
-export const protocol: Protocol<ChannelState> = (_ps: ChannelState) => Promise.resolve(right(none));
+import {Protocol, ChannelState, stage, ProtocolResult} from './state';
+import {protocol as fundingProtocol} from './depositing';
+
+export type DirectFundingState = ChannelState & {minimalOutcome: SimpleAllocation};
```

Comment:
This assumes that the  `minimalOutcome` would come from the application request or an Objective shared from the opponent.

Reply:
Should it be a `MixedAllocation`? IE. should we prepare for multi-token channels?

Reply:
> Could we get a direct funding protocol checked in that is partially implemented? EG. it can ignore depositing, and just sign pre- and post-fund-setup states?

I think it would be pretty easy to get a partially implemented version checked in. However it would be dependent on implementation of `Action` and the basic runloop ( would that be in scope of #2311 ?)